spring-cloud-gcp-data-spanner/src/main/java/org/springframework/cloud/gcp/data/spanner/core/convert/MappingSpannerWriteConverter.java

```diff
@@ -156,19 +160,36 @@ private void writeProperty(WriteBuilder sink, PersistentPropertyAccessor accesso
 		Class<?> propertyType = property.getType();
 		ValueBinder<WriteBuilder> valueBinder = sink.set(property.getColumnName());
 
-		boolean valueSet;
+		boolean valueSet = false;
 
 		/*
-		 * Due to type erasure, binder methods for Iterable properties must be manually specified.
-		 * ByteArray must be excluded since it implements Iterable, but is also explicitly
-		 * supported by spanner.
+		 * Due to type erasure, binder methods for Iterable properties must be manually
+		 * specified. ByteArray must be excluded since it implements Iterable, but is also
+		 * explicitly supported by spanner.
+		 *
+		 * Also due to type erasure, custom conversions cannot be used for iterables
+		 * because different conversions between lists of different types cannot be
+		 * distinguished.
```

Comment:
I see, the problem is that targetType is just a list, so you can't just simply call `canConvert` on it, right?
It would be nice if we could just assume lists should be of a single type of element defined by `@ColumnInnerType` and then have a "map(x -> convert(x))" function to map...for that maybe we could introduce a `source` and `targetType` for the `@ColumnInnerType` annotation?

Reply:
I agree that we could make the assumption that all items in the list would work for a single conversion function, however there are cases where we just don't have enough type information to do anything:
if the user specifies a conversion from a type Car to  List<String> and a different conversion from Car to List<Integer> (perhaps capturing different parts of the information in a Car), then our matching would just see two converters from Car to List and we wouldn't know which converter is correct (or perhaps always choose the first one that matches and never the other one). 

However, I hesitate to include inner-type annotations and information in this custom conversion picture. 
Let's say in our POJO we have a field List<List<Character>> and the user would like that to be stored in Spanner as List<String> (it's weird, but conceivable).  That would mean we need a pair of converters from List<Character> to String and back, but because of type erasure that would be just List-to-String, which means we're back at square one. 

I think we can overcome this by making our own type EnhancedConverter that holds the inner type. I think that could be the next PR as well.